### PR TITLE
feat(serve): implement open default browser option for ng serve

### LINKED ISSUE #1081: Executing "ng serve" or "npm start" command browser is not opening automatically
1. OS?  
Window 7

2. Versions?
angular-cli: 0.0.39
node: 5.4.1
os: win32 x64

I have generated a project using angular-cli by following steps given in installation section. Project is running right now. But i have to open browser manually by typing http://localhost:4200/ into browser.

Could it be possible like when user will execute "ng serve" or "npm start" command, browser will open automatically?

Please let me know if it is possible or give me any solution for this.

Thanks in advance.

-Deepak

### PR DESCRIPTION
**Pull request by dzonatan**
Sometimes it's really handy to have auto browser opening when serving app.
This PR implements that by `--open` option for `ng serve` command.

Closes #1081

### CODE REVIEW

File: packages/angular-cli/tasks/serve-webpack.ts [hunk 1]
```diff
@@ -78,6 +79,10 @@ export default Task.extend({
           console.error(err.stack || err);
           if (err.details) { console.error(err.details); }
           reject(err.details);
+        } else {
+          if(commandOptions.open) {
+            opn(`http://${commandOptions.host}:${commandOptions.port}/`);
```
Comment (graingert) [inline packages/angular-cli/tasks/serve-webpack.ts]:
```js
import url from 'url';
...
const { open, host, port } = commandOptions;
if (open) {
  opn(url.format({ protocol: 'http:', host, port }));
}
```

Comment (dzonatan) [inline packages/angular-cli/tasks/serve-webpack.ts]: Are there any pros of that url formatter?
I was trying to make it same as [webpack dev server binary](https://github.com/webpack/webpack-dev-server/blob/master/bin/webpack-dev-server.js#L296) do.

Comment (graingert) [inline packages/angular-cli/tasks/serve-webpack.ts]: It's built in and escapes funky stuff like ipv6 addresses properly

File: packages/angular-cli/commands/serve.ts [hunk 2]
```diff
@@ -79,7 +80,13 @@ const ServeCommand = Command.extend({
     { name: 'ssl',                  type: Boolean, default: false },
     { name: 'ssl-key',              type: String,  default: 'ssl/server.key' },
     { name: 'ssl-cert',             type: String,  default: 'ssl/server.crt' },
-    { name: 'aot',                  type: Boolean, default: false }
+    { name: 'aot',                  type: Boolean, default: false },
+    {
+      name: 'open',
+      description: 'Opens the url in default browser',
+      type: Boolean,
+      default: false
```
Comment (filipesilva) [inline packages/angular-cli/commands/serve.ts]: Can you add the `o` alias?
```
aliases: ['o']
```

**Comment by filipesilva** [comment]
This is nice, thanks! I added the request for an alias, I think it would help. People could then do `ng serve -o`.

File: packages/angular-cli/commands/serve.ts (hunk 2 above)
Comment (dzonatan) [inline packages/angular-cli/commands/serve.ts]: done!

File: packages/angular-cli/tasks/serve-webpack.ts [hunk 3]
```diff
@@ -78,6 +80,11 @@ export default Task.extend({
           console.error(err.stack || err);
           if (err.details) { console.error(err.details); }
           reject(err.details);
+        } else {
+          const { open, host, port } = commandOptions;
+          if (open) {
+            opn(url.format({ protocol: 'http', hostname: host, port: port.toString() }));
```
Comment (graingert) [inline packages/angular-cli/tasks/serve-webpack.ts]: nice

**Comment by Splaktar** [comment]
Thank you!

**Pull request closed without merging**